File: src/main/java/no/nav/familie/ks/oppslag/egenansatt/EgenAnsattService.java
package no.nav.familie.ks.oppslag.egenansatt;

import no.nav.familie.ks.oppslag.egenansatt.internal.EgenAnsattConsumer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class EgenAnsattService {

    private EgenAnsattConsumer egenAnsattConsumer;

    @Autowired
    EgenAnsattService (EgenAnsattConsumer egenAnsattConsumer) {
        this.egenAnsattConsumer = egenAnsattConsumer;
    }

    public boolean erEgenAnsatt(String fnr) {
        return egenAnsattConsumer.erEgenAnsatt(fnr);
    }
}

File: src/main/java/no/nav/familie/integrasjoner/egenansatt/EgenAnsattService.kt
package no.nav.familie.integrasjoner.egenansatt

import no.nav.familie.integrasjoner.client.rest.EgenAnsattRestClient
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Service

@Service
class EgenAnsattService(private val egenAnsattRestClient: EgenAnsattRestClient) {
    @Cacheable("erEgenAnsatt")
    fun erEgenAnsatt(personIdent: String): Boolean = egenAnsattRestClient.erEgenAnsatt(personIdent)

    fun erEgenAnsatt(personIdenter: Set<String>): Map<String, Boolean> = egenAnsattRestClient.erEgenAnsatt(personIdenter)
}
